prototype_source/quantization_in_pytorch_2_0_export_tutorial.rst

```diff
@@ -0,0 +1,341 @@
+   or `FakeQuantizeBase <https://github.com/pytorch/pytorch/blob/07104ca99c9d297975270fb58fda786e60b49b38/torch/ao/quantization/fake_quantize.py#L60>`__)
+   as input. From each ``ObserverOrFakeQuantize`` object, user can get the ``scale``, ``zero point`` value. Combine these values together,
+   user can define its heuristic about how to derive new ``scale``, ``zero point`` value.
+-  Step 2: Define ``DerivedQuantizationSpec`` obejct, it accepts inputs of: list of ``EdgeOrNode`` objects.
+   The observer corresponding to each ``EdgeOrNode`` object will be passed into the ``derive_qparams_fn`` function;
+   ``derive_qparams_fn`` function; several other quantization parameters such as ``dtype``, ``qscheme``.
+-  Step 3: Annotate the inputs and output of this conv node with ``QuantizationAnnotation``.
+
+::
+
+    def derive_qparams_fn(obs_or_fqs: List[ObserverOrFakeQuantize]) -> Tuple[Tensor, Tensor]:
+        assert len(obs_or_fqs) == 2, \
+            "Expecting two obs/fqs, one for activation and one for weight, got: {}".format(len(obs_or_fq))
+        act_obs_or_fq = obs_or_fqs[0]
+        weight_obs_or_fq = obs_or_fqs[1]
+        act_scale, act_zp = act_obs_or_fq.calculate_qparams()
+        weight_scale, weight_zp = weight_obs_or_fq.calculate_qparams()
+        return torch.tensor([act_scale * weight_scale]).to(torch.float32), torch.tensor([0]).to(torch.int32)
+
+    bias_qspec = DerivedQuantizationSpec(
+        derived_from=[(input_act, node), (weight, node)],
+        derive_qparams_fn=derive_qparams_fn,
+        dtype=torch.int32,
+        quant_min=-2**31,
+        quant_max=2**31 - 1,
+        qscheme=torch.per_tensor_symmetric,
+    )
+    input_qspec_map = {input_act: act_quantization_spec, weight: weight_quantization_spec, bias: bias_qspec}
+    node.meta["quantization_annotation"] = QuantizationAnnotation(
+        input_qspec_map=input_qspec_map,
+        output_qspec=act_quantization_spec,
+        _annotated=True,
+    )
+
+5. A Toy Example with Resnet18 
+--------------------------------------------------------
+
+After above annotation methods defined with ``QuantizationAnnotation API``, we can now put them together to construct a ``BackendQuantizer``
+and run a `toy example <https://gist.github.com/leslie-fang-intel/b78ed682aa9b54d2608285c5a4897cfc>`__
+with ``Torchvision Resnet18``. To better understand the final example, here are some basic concepts which are used:
+
+- `QuantizationConfig <https://github.com/pytorch/pytorch/blob/73fd7235ad25ff061c087fa4bafc6e8df4d9c299/torch/ao/quantization/_pt2e/quantizer/quantizer.py#L103-L109>`__
+  consists of ``QuantizationSpec`` for activation, weight, and bias separately.
+- When annotating the model, methods of
+  `get_act_qspec <https://github.com/pytorch/pytorch/blob/73fd7235ad25ff061c087fa4bafc6e8df4d9c299/torch/ao/quantization/_pt2e/quantizer/utils.py#L9>`__,
```

Comment:
nit: this is now changed to https://github.com/pytorch/pytorch/blob/main/torch/ao/quantization/_pt2e/quantizer/utils.py#L10 and https://github.com/pytorch/pytorch/blob/main/torch/ao/quantization/_pt2e/quantizer/utils.py#L23

Reply:
Thanks and the link are updated.